src/utils/smartParser.ts

```diff
@@ -207,25 +206,46 @@ export const smartParser = (filePath: string, fileContent: any) => {
           } else if (fileDepth === 3) {
             const nestedArrayName = String(Object.keys(fileContent));
 
-            const nestedDataSelector = (parsingModel.parsingmodel as any)[
-              filePathModel
-            ]['file_structure_properties']['nested_data_point_selector']; // Select the right nested array name with nestedDataSelector property to parse the object.
-
-            for (let j = 0; j < fileContent[nestedArrayName].length; j++) {
-              for (
-                let k = 0;
-                k < fileContent[nestedArrayName][j][nestedDataSelector].length;
-                k++
-              ) {
-                const indivArray = [];
-                for (let l = 0; l < ObjectPropertiesName.length; l++) {
-                  indivArray.push(
-                    (parsingModel.parsingmodel as any)[filePathModel][
-                      nestedArrayName
-                    ]['entries'][l][ObjectPropertiesName[l]]
-                  );
+            if ((parsingModel.parsingmodel as any)[filePathModel]['file_structure_properties']['nested_data_point_selector'] != ''){
+              const nestedDataSelector = (parsingModel.parsingmodel as any)[
+                filePathModel
+              ]['file_structure_properties']['nested_data_point_selector']; // Select the right nested array name with nestedDataSelector property to parse the object.
+  
+              for (let j = 0; j < fileContent[nestedArrayName].length; j++) {
+                for (
+                  let k = 0;
+                  k < fileContent[nestedArrayName][j][nestedDataSelector].length;
+                  k++
+                ) {
+                  const indivArray = [];
+                  for (let l = 0; l < ObjectPropertiesName.length; l++) {
```

Comment:
SI on prend l'exemple des fichiers dans le dossier `fake-data`, tu as 2 itérations et donc 2 data points un pour `name` et un pour `displayName` ? Même s'il y a plusieurs variables au sein de `name`?

Reply:
Oui, j'ai fait le choix de ne pas faire une itération supplémentaire pour aller chercher : `givenName`, `familyName`, `formattedName`, je trouvais ça redondant de considérer chacune de ces informations comme des data points séparés. Qu'est-ce que tu en penses toi ?